contracts/RewardPool.sol

```diff
@@ -92,55 +93,56 @@ contract RewardPool is Initializable, Versionable, Ownable {
     return true;
   }
 
-  function claim(
-    address rewardProgramID,
-    address payableToken,
-    uint256 amount,
-    bytes calldata proof
-  ) external returns (bool) {
-    require(amount > 0, "Cannot claim non-positive amount");
-    address rewardSafeOwner = RewardManager(rewardManager).getRewardSafeOwner(
-      msg.sender
-    );
-    require(
-      RewardManager(rewardManager).isValidRewardSafe(
-        msg.sender,
-        rewardProgramID
-      ),
-      "can only withdraw for safe registered on reward program"
+  function valid(bytes memory leaf, bytes32[] memory proof)
+    public
+    view
+    returns (bool)
+  {
+    (, uint256 paymentCycleNumber, , , , , ) = abi.decode(
```

Comment:
Sorry to be a pain, but now that I see this I hate it even more, it feels like a big footgun. If the comments weren't pulled out by the annoying formatter then it would be better but actually still pretty error prone.

I think it would be really nice to have a single source of truth for "how to extract the `paymentCycleNumber` from leaf bytes, so a function `extractPaymentCycleNumber(bytes memory leaf) returns (uint256)` etc

Adding a bunch of getters like this in this contract will bloat bytes and not be reusable. I think that this is a great use case for a library with external view functions, which is deployed at it's own address on the blockchain and therefore doesn't count towards this contract's bytesize limit.

It's a bit nitpicky I know, but I feel having a SSOT for decoding this that can be used here and in other contracts will be a prudent way to do this and less error-prone as we use it more often.